### ISSUE TITLE
How to use the LRN layer with video-caffe?

### ISSUE THREAD

**Issue by livst** [chuckcho/video-caffe#60]
Hi,

How can I use the Local Response Normalization (LRN) layer written below, with this caffe version?

layer {
  name: "norm1"
  type: "LRN"
  bottom: "conv1"
  top: "norm1"
  lrn_param {
    local_size: 5
    alpha: 0.0001
    beta: 0.75
  }
}

When i use it as  above i get an error about the size (5 vs 4).
Thank you for your assistance,
livst

**Comment by chuckcho** [comment]
@livst thanks for bringing this up. will take a look in my spare time.

**Comment by pelun** [comment]
When I run make runtest, it shows:
.build_release/tools/caffe
caffe: command line brew
usage: caffe <command> <args>

commands:
  train           train or finetune a model
  test            score a model
  device_query    show GPU diagnostic information
  time            benchmark model execution time

  Flags from tools/caffe.cpp:
    -gpu (Optional; run in GPU mode on given device IDs separated by ','.Use
      '-gpu all' to run on all available GPUs. The effective training batch
      size is multiplied by the number of devices.) type: string default: ""
    -iterations (The number of iterations to run.) type: int32 default: 50
    -level (Optional; network level.) type: int32 default: 0
    -model (The model definition protocol buffer text file.) type: string
      default: ""
    -phase (Optional; network phase (TRAIN or TEST). Only used for 'time'.)
      type: string default: ""
    -sighup_effect (Optional; action to take when a SIGHUP signal is received:
      snapshot, stop or none.) type: string default: "snapshot"
    -sigint_effect (Optional; action to take when a SIGINT signal is received:
      snapshot, stop or none.) type: string default: "stop"
    -snapshot (Optional; the snapshot solver state to resume training.)
      type: string default: ""
    -solver (The solver definition protocol buffer text file.) type: string
      default: ""
    -stage (Optional; network stages (not to be confused with phase), separated
      by ','.) type: string default: ""
    -weights (Optional; the pretrained weights to initialize finetuning,
      separated by ','. Cannot be set simultaneously with snapshot.)
      type: string default: ""
.build_release/test/test_all.testbin 0 --gtest_shuffle 
NVIDIA: no NVIDIA devices found
Cuda number of devices: 0
Setting to use device 0
Current device id: 0
Current device name: 
Note: Randomizing tests' orders with a seed of 71371 .
[==========] Running 2125 tests from 283 test cases.
[----------] Global test environment set-up.
[----------] 27 tests from ReductionLayerTest/0, where TypeParam = caffe::CPUDevice<float>
[ RUN      ] ReductionLayerTest/0.TestMeanCoeffGradientAxis1
NVIDIA: no NVIDIA devices found
E1130 06:43:11.260094  4818 common.cpp:113] Cannot create Cublas handle. Cublas won't be available.
NVIDIA: no NVIDIA devices found
E1130 06:43:11.264824  4818 common.cpp:120] Cannot create Curand generator. Curand won't be available.
E1130 06:43:11.264919  4818 common.cpp:140] Curand not available. Skipping setting the curand seed.
[       OK ] ReductionLayerTest/0.TestMeanCoeffGradientAxis1 (13 ms)
[ RUN      ] ReductionLayerTest/0.TestSum
[       OK ] ReductionLayerTest/0.TestSum (0 ms)
[ RUN      ] ReductionLayerTest/0.TestSetUpWithAxis2
[       OK ] ReductionLayerTest/0.TestSetUpWithAxis2 (0 ms)
[ RUN      ] ReductionLayerTest/0.TestSumCoeffAxis1Gradient
[       OK ] ReductionLayerTest/0.TestSumCoeffAxis1Gradient (1 ms)
[ RUN      ] ReductionLayerTest/0.TestMean
[       OK ] ReductionLayerTest/0.TestMean (0 ms)
[ RUN      ] ReductionLayerTest/0.TestSumGradient
[       OK ] ReductionLayerTest/0.TestSumGradient (1 ms)
[ RUN      ] ReductionLayerTest/0.TestMeanGradient
[       OK ] ReductionLayerTest/0.TestMeanGradient (0 ms)
[ RUN      ] ReductionLayerTest/0.TestSumOfSquaresCoeffGradient
[       OK ] ReductionLayerTest/0.TestSumOfSquaresCoeffGradient (1 ms)
[ RUN      ] ReductionLayerTest/0.TestSumCoeff
[       OK ] ReductionLayerTest/0.TestSumCoeff (0 ms)
[ RUN      ] ReductionLayerTest/0.TestAbsSum
[       OK ] ReductionLayerTest/0.TestAbsSum (14 ms)
[ RUN      ] ReductionLayerTest/0.TestAbsSumCoeffAxis1Gradient
[       OK ] ReductionLayerTest/0.TestAbsSumCoeffAxis1Gradient (1 ms)
[ RUN      ] ReductionLayerTest/0.TestSumOfSquaresCoeffAxis1
[       OK ] ReductionLayerTest/0.TestSumOfSquaresCoeffAxis1 (0 ms)
[ RUN      ] ReductionLayerTest/0.TestAbsSumCoeff
[       OK ] ReductionLayerTest/0.TestAbsSumCoeff (0 ms)
[ RUN      ] ReductionLayerTest/0.TestSumCoeffGradient
[       OK ] ReductionLayerTest/0.TestSumCoeffGradient (1 ms)
[ RUN      ] ReductionLayerTest/0.TestMeanCoeff
[       OK ] ReductionLayerTest/0.TestMeanCoeff (0 ms)
[ RUN      ] ReductionLayerTest/0.TestAbsSumGradient
[       OK ] ReductionLayerTest/0.TestAbsSumGradient (1 ms)
[ RUN      ] ReductionLayerTest/0.TestSumOfSquaresCoeff
[       OK ] ReductionLayerTest/0.TestSumOfSquaresCoeff (0 ms)
[ RUN      ] ReductionLayerTest/0.TestMeanCoeffGradient
[       OK ] ReductionLayerTest/0.TestMeanCoeffGradient (1 ms)
[ RUN      ] ReductionLayerTest/0.TestSumOfSquares
[       OK ] ReductionLayerTest/0.TestSumOfSquares (0 ms)
[ RUN      ] ReductionLayerTest/0.TestSetUpWithAxis1
[       OK ] ReductionLayerTest/0.TestSetUpWithAxis1 (0 ms)
[ RUN      ] ReductionLayerTest/0.TestSumOfSquaresCoeffAxis1Gradient
[       OK ] ReductionLayerTest/0.TestSumOfSquaresCoeffAxis1Gradient (1 ms)
[ RUN      ] ReductionLayerTest/0.TestAbsSumCoeffAxis1
[       OK ] ReductionLayerTest/0.TestAbsSumCoeffAxis1 (0 ms)
[ RUN      ] ReductionLayerTest/0.TestAbsSumCoeffGradient
[       OK ] ReductionLayerTest/0.TestAbsSumCoeffGradient (0 ms)
[ RUN      ] ReductionLayerTest/0.TestSumCoeffAxis1
[       OK ] ReductionLayerTest/0.TestSumCoeffAxis1 (0 ms)
[ RUN      ] ReductionLayerTest/0.TestSumOfSquaresGradient
[       OK ] ReductionLayerTest/0.TestSumOfSquaresGradient (1 ms)
[ RUN      ] ReductionLayerTest/0.TestMeanCoeffAxis1
[       OK ] ReductionLayerTest/0.TestMeanCoeffAxis1 (0 ms)
[ RUN      ] ReductionLayerTest/0.TestSetUp
[       OK ] ReductionLayerTest/0.TestSetUp (0 ms)
[----------] 27 tests from ReductionLayerTest/0 (36 ms total)

[----------] 2 tests from SoftmaxLayerTest/2, where TypeParam = caffe::GPUDevice<float>
[ RUN      ] SoftmaxLayerTest/2.TestForward
F1130 06:43:11.289638  4818 syncedmem.hpp:18] Check failed: error == cudaSuccess (30 vs. 0)  unknown error
*** Check failure stack trace: ***
    @     0x7fab8509b5cd  google::LogMessage::Fail()
    @     0x7fab8509d433  google::LogMessage::SendToLog()
    @     0x7fab8509b15b  google::LogMessage::Flush()
    @     0x7fab8509de1e  google::LogMessageFatal::~LogMessageFatal()
    @     0x7fab82aa5ee1  caffe::SyncedMemory::mutable_cpu_data()
    @     0x7fab82ad39f1  caffe::Blob<>::Reshape()
    @     0x7fab82ad3eb9  caffe::Blob<>::Reshape()
    @     0x7fab82ad4070  caffe::Blob<>::Blob()
    @           0x748e6c  caffe::SoftmaxLayerTest<>::SoftmaxLayerTest()
    @           0x74905b  testing::internal::TestFactoryImpl<>::CreateTest()
    @           0x962ce3  testing::internal::HandleExceptionsInMethodIfSupported<>()
    @           0x95c2a3  testing::TestInfo::Run()
    @           0x95c405  testing::TestCase::Run()
    @           0x95d6df  testing::internal::UnitTestImpl::RunAllTests()
    @           0x95da03  testing::UnitTest::Run()
    @           0x46fb6d  main
    @     0x7fab81e32830  __libc_start_main
    @           0x4779e9  _start
    @              (nil)  (unknown)
Makefile:526: recipe for target 'runtest' failed
make: *** [runtest] Aborted (core dumped)

C

**Comment by chuckcho** [comment]
I see `NVIDIA: no NVIDIA devices found`. do you have GPU(s)? if so, did you install CUDA / CuDNN?

**Comment by pelun** [comment]
Thanks.I did install cuda 8.0 and cudnn 5.1.

**Comment by chuckcho** [comment]
It looks to me CUDA/CuDNN is not detected. Can you try [refactor branch](https://github.com/chuckcho/video-caffe/tree/refactor)?

**Comment by pelun** [comment]
Thanks. I see. Can you tell me details? I haven't understood #refactor branch.

**Comment by chuckcho** [comment]
`git checkout refactor` in command-line or download `https://github.com/chuckcho/video-caffe/archive/refactor.zip`

**Comment by pelun** [comment]
Thanks for your help. I will try!

**Comment by pelun** [comment]
@chuckcho . I build with refactor branch. So, my problems haven't resolved.

**Comment by chuckcho** [comment]
@livst is this issue still relevant? if not, i'll close. thx.
